include/copc-lib/las/point.hpp

```diff
@@ -1,597 +1,590 @@
 #ifndef COPCLIB_LAS_POINT_H_
 #define COPCLIB_LAS_POINT_H_
 
-#include <cstdint>
-#include <iostream>
 #include <sstream>
-#include <stdexcept>
 #include <vector>
 
-#include <copc-lib/las/internal/extra_bytes.hpp>
-#include <copc-lib/las/internal/gps_time.hpp>
-#include <copc-lib/las/internal/nir.hpp>
-#include <copc-lib/las/internal/point_10.hpp>
-#include <copc-lib/las/internal/point_14.hpp>
-#include <copc-lib/las/internal/rgb.hpp>
-#include <copc-lib/las/internal/utils.hpp>
-
 namespace copc::las
 {
 
 class Point
 {
   public:
-    Point(const uint8_t &point_format_id)
-    {
-        point_10_ = nullptr;
-        point_14_ = nullptr;
-        rgb_ = nullptr;
-        gps_time_ = nullptr;
-        nir_ = nullptr;
+    Point() = default;
 
-        switch (point_format_id)
-        {
-        case 0:
-            point_10_ = new internal::Point10();
-            break;
-        case 1:
-            point_10_ = new internal::Point10();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 2:
-            point_10_ = new internal::Point10();
-            rgb_ = new internal::Rgb();
-            break;
-        case 3:
-            point_10_ = new internal::Point10();
-            rgb_ = new internal::Rgb();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 4:
-            point_10_ = new internal::Point10();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 5:
-            point_10_ = new internal::Point10();
-            rgb_ = new internal::Rgb();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 6:
-            point_14_ = new internal::Point14();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 7:
-            point_14_ = new internal::Point14();
-            rgb_ = new internal::Rgb();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 8:
-            point_14_ = new internal::Point14();
-            rgb_ = new internal::Rgb();
-            gps_time_ = new internal::GpsTime();
-            nir_ = new internal::Nir();
-            break;
-        case 9:
-            point_14_ = new internal::Point14();
-            gps_time_ = new internal::GpsTime();
-            break;
-        case 10:
-            point_14_ = new internal::Point14();
-            rgb_ = new internal::Rgb();
-            gps_time_ = new internal::GpsTime();
-            nir_ = new internal::Nir();
-            break;
-        default:
-            throw std::runtime_error("Point format must be 0-10");
-        }
+    void Unpack(std::istream &in_stream, const uint16_t &point_format_id, const uint16_t &point_record_length);
```

Comment:
I think this should be a static function that returns a Point

Reply:
So separate from the Point class?

Reply:
It can be within the point class. But it should be a static function that returns a point object. See here: https://github.com/RockRobotic/copc-lib/blob/writer/include/copc-lib/hierarchy/entry.hpp